configuration/pih/statusData/hiv/lastPregnancyStatus.sql

```diff
@@ -1,25 +1,13 @@
-# This retrieves all of the data points needed to provide information as to
-# the last pregnancy status
-
 # Gender
 select gender into @gender from person where person_id = @patientId;
 
-# Pregnancy Status
-select e.encounter_id, o.value_coded into @pregnantEncounterId, @pregnantValueCoded
-from obs o
-inner join encounter e on o.encounter_id = e.encounter_id
-where o.person_id = @patientId
-and o.concept_id = concept_from_mapping('CIEL', '5272')
-and o.voided = 0
-and e.voided = 0
-order by e.encounter_datetime desc
-limit 1
-;
 
-select obs_value_datetime(@pregnantEncounterId, 'CIEL', '5596') into @edd;
+# Estimated delivery date
+select o.value_datetime into @edd 
+from obs o
+where o.obs_id = latest_obs(@patientId, 'PIH', '5596')
+and o.voided = 0;
 
 select
-    @gender as gender,
-    if (@pregnantValueCoded = concept_from_mapping('CIEL', '1065'), true, false) as pregnant,
-    if (@pregnantValueCoded = concept_from_mapping('CIEL', '1065'), date(@edd), null) as dueDate
-;
+  @gender as gender,
+    if(@edd > now(),date(@edd),null) as estimatedDeliveryDate;
```

Comment:
This could become tricky to calculate, but, what if the baby is overdue? I think this assumes that once you reach and pass the delivery date, the patient is no longer pregnant, which isn't the case if the baby is delivered late?

Reply:
You're right, @mogoodrich . It's a tricky calculation. 
you can't base the calculation on the estimated date of delivery. The clinician or the patient may not know the last date of menstruation.

I think we need to consider other parameters for the calculation.